### PR TITLE
Update setting-default-values-with-short-circuiting.md

### PR DESCRIPTION
**Pull request by nem035**

### CODE REVIEW

**Comment by enki-hq** [comment]
### Insight previews:
- [ ] [enkidevs/curriculum/javascript/core/functions-i/setting-default-values-with-short-circuiting.md](https://app.enkipro.com/#/public/preview-insight/enkidevs/curriculum/javascript/core/functions-i/setting-default-values-with-short-circuiting.md?ref=c5131af96d94153b862d7bdfc939ec1d33e39e79)

File: javascript/core/functions-i/setting-default-values-with-short-circuiting.md [hunk 1]
```diff
@@ -46,17 +29,38 @@ aspects:
...
 
 ```javascript
-function documentTitle(theTitle) {
-  theTitle = theTitle || "Untitled Doc";
+function title(t) {
+  t = t || "Untitled";
+}
+```
+
+The above will return `"Untitled"` for `t` if the provided value for `t` is falsy[1].
```
Comment (nem035) [inline javascript/core/functions-i/setting-default-values-with-short-circuiting.md:47]:
```suggestion
The above will return `"Untitled"` for `t` if the provided value for `t` is falsy
```

File: javascript/core/functions-i/setting-default-values-with-short-circuiting.md [hunk 2]
```diff
@@ -46,17 +29,38 @@ aspects:
...
+
+Sometimes this isn't desired. If an empty title is allowed, we would (incorrectly) get `"Untitled"` instead (because an empty string `''` is falsy).
+
+A more correct approach could be to use default parameters to short-circuit the default value only if the provided one is `undefined`.
```
Comment (nem035) [inline javascript/core/functions-i/setting-default-values-with-short-circuiting.md:51]:
```suggestion
A better approach would be to use default parameters to use the default value only if the provided one is `undefined`.
```

File: javascript/core/functions-i/setting-default-values-with-short-circuiting.md [hunk 3]
```diff
@@ -46,17 +29,38 @@ aspects:
...
+
+```javascript
+function title(t = "Untitled") {
+  
+}
+```
+
+If we wanted additional safety against both `undefined` and `null` (but allow all other values), we could use the nullish coalescing operator:
```
Comment (nem035) [inline javascript/core/functions-i/setting-default-values-with-short-circuiting.md:59]:
```suggestion
If we wanted additional safety against both `undefined` and `null` (but allow all other values), we'd use the nullish coalescing operator `??`:
```

**Pull request merged**